### PR TITLE
child_process: clone spawn options argument

### PR DESCRIPTION
**Pull request by jasnell**
spawnSync() modifies the options argument. This commit makes
a copy of options before any modifications occur.

### CODE REVIEW

**Comment by jasnell** [comment]
Per: https://github.com/joyent/node/issues/9158

**Comment by jasnell** [comment]
cc: @joyent/node-coreteam ... this was a pull over of a change @cjihrig made over in io.js land if I recall correctly.

File: lib/child_process.js [hunk 1]
```diff
@@ -960,6 +960,7 @@ function normalizeSpawnArguments(file /*, args, options*/) {
   else if (!util.isObject(options))
     throw new TypeError('options argument must be an object');
 
+  options = util._extend({}, options);
```
Comment (trevnorris) [inline lib/child_process.js]: is this basically to shallow clone `options`?

**Comment by jasnell** [comment]
Effectively yes. The intent is to isolate the changes spawnSync makes to options so that those don't bleed out.

**Comment by trevnorris** [comment]
cool. I'm fine with this.

**Comment by jasnell** [comment]
Landed in b5a0abc

**Pull request closed without merging**